src/Overlay.js

```diff
@@ -27,15 +33,36 @@ class Overlay extends React.Component {
     return null
   }
 
+  componentDidMount() {
+    this.setState({ target: this.getTarget() })
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props === prevProps) return
```

Comment:
it's really ok/meaningful to do a ref equality check on all props?

shouldn't the props _always_ change?

Reply:
It's a heuristic for "just state changed"  which is the only time they don't change